spec/api-desktop-capturer-spec.js

```diff
@@ -91,12 +91,145 @@ describe('desktopCapturer', () => {
+    const MAX_WIN = 4
+    const { BrowserWindow } = remote
+    const mainWindow = remote.getCurrentWindow()
+    const wList = [mainWindow]
+    // Add MAX_WIN-1 more window so we have MAX_WIN in total.
+    for (let i = 0; i < MAX_WIN - 1; i++) {
+      const w = new BrowserWindow({ show: true, width: 100, height: 100 })
+      wList.push(w)
+      w.loadURL('about:blank')
+      w.show()
+    }
+    expect(wList.length).to.equal(MAX_WIN)
+
+    // Show and focus all the windows.
+    wList.forEach(async (w) => {
+      const wFocused = emittedOnce(w, 'focus')
+      w.focus()
+      await wFocused
+    })
+    // At this point our windows should be showing from bottom to top.
+
+    // DesktopCapturer.getSources() returns sources sorted from foreground to
+    // background, i.e. top to bottom.
+    let sources = await desktopCapturer.getSources({
+      types: ['window'],
+      thumbnailSize: { width: 0, height: 0 }
+    })
+
+    // TODO(julien.isorce): investigate why |sources| is empty on the linux
+    // bots while it is not on my workstation, as expected, with and without
+    // the --ci parameter.
+    if (process.platform === 'linux' && sources.length === 0) {
+      wList.forEach((w) => {
+        if (w !== mainWindow) {
+          w.destroy()
+        }
+      })
+      return
+    }
+
+    expect(sources).to.be.an('array').that.is.not.empty()
+    expect(sources.length).to.gte(MAX_WIN)
+
+    // Only keep our windows, they must be in the MAX_WIN first windows.
+    sources.splice(MAX_WIN, sources.length - MAX_WIN)
+    expect(sources.length).to.equal(MAX_WIN)
+    expect(sources.length).to.equal(wList.length)
+
+    // Check that the sources and wList are sorted in the reverse order.
+    let canGofurther = true
+    const wListReversed = wList.slice(0).reverse()
+    sources.forEach((source, index) => {
+      if (source.id !== wListReversed[index].getMediaSourceId()) {
+        canGofurther = false
+      }
+    })
+    if (!canGofurther) {
+      // Skip remaining checks because either focus or window placement are
+      // not reliable in the running test environment. So there is no point
+      // to go further to test moveAbove as requirements are not met.
+      wList.forEach((w) => {
+        if (w !== mainWindow) {
+          w.destroy()
+        }
+      })
+      return
+    }
+
+    // Do the real work, i.e. move each window above the next one so that
+    // wList is sorted from foreground to background.
+    wList.forEach(async (w, index) => {
+      if (index < (wList.length - 1)) {
+        const wNext = wList[index + 1]
+        w.moveAbove(wNext.getMediaSourceId())
+      }
+    })
+
+    sources = await desktopCapturer.getSources({
+      types: ['window'],
+      thumbnailSize: { width: 0, height: 0 }
+    })
+    // Only keep our windows again.
+    sources.splice(MAX_WIN, sources.length - MAX_WIN)
+    expect(sources.length).to.equal(MAX_WIN)
+    expect(sources.length).to.equal(wList.length)
```

Comment:
perhaps `sources = sources.filter(source => ourWindows.contains(source.id))` would be more robust? the relative ordering of our windows should be consistent even if other windows manage to get in the way.

Reply:
contains does not exist and includes did not work because of the `.id` so I left it like this